### PR TITLE
nmap: 7.80 -> 7.91

### PR DESCRIPTION
**Pull request by holzingk**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Ma27** [comment]
`nmap-unfree` is already at 7.91, `nmap` should stay at 4.80 for now due to licensing issues.

**Comment by nextl00p** [comment]
See also: #105119 and last paragraph of https://nmap.org/npsl/

TL;DR: They re-licensed version 7.90 and 7.91 under the old license. So I believe at least for this particular version the licensing issues are resolved.

**Comment by Ma27** [comment]
Fair enough, sorry. Reopening.

I guess in this case we can drop `nmap-unfree` though and make it an alias to `nmap`?

**Comment by nextl00p** [comment]
> I guess in this case we can drop nmap-unfree though and make it an alias to nmap?

As far as I understand the licensing remains unclear for future releases. Maybe just leave those two in parallel for now and let's see whether `nmap-unfree` will be required.

**Comment by Ma27** [comment]
I'm confused why we even have copied one expression there. Aliasing should be fine for now, if we license thing re-occurs, we can only make the new version available at `nmap-unfree` by removing the alias.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118158` at e12bbb6a run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built successfully:</summary>
  <ul>
    <li>crowbar</li>
    <li>gnome3.gnome-nettool</li>
    <li>nmap</li>
    <li>nmap-graphical</li>
    <li>python38Packages.python-nmap</li>
    <li>python39Packages.python-nmap</li>
    <li>rmount</li>
    <li>rustscan</li>
  </ul>
</details>

**Comment by fabaff** [comment]
With the statement on the website, we no longer need `nmap-unfree`. I'm not sure that one sentence or paragraph without date or other reference, five months after the release, qualifies as re-licensing. The license file that is part of the 7.91 source says NPSL and the file headers also says NPSL. The source for 7.91 doesn't contain any details about the previous license or that's allowed to use it under GPLv2.

**Comment by nextl00p** [comment]
Also Debian sid included nmap 7.91 again.

It is true that the sources still mention the NPSL. GPLv2 is just an alternative license that we are allowed to use.

**Comment by fabaff** [comment]
I assume that the Debian legal team has reviewed the current situation and if they agrees that the available statements are enough, it's fine for me.

**Comment by packetizeme** [comment]
Worth noting that nmap 7.80 is currently broken on MacOS:
```
Starting Nmap 7.80 ( https://nmap.org ) at 2021-05-25 10:00 PDT
Assertion failed: (res > 7), function nsock_library_initialize, file nsock_pool.c, line 307.
```

This is fixed in 7.91 per https://github.com/nmap/nmap/issues/2081

**Comment by SuperSandro2000** [comment]
> I assume that the Debian legal team has reviewed the current situation and if they agrees that the available statements are enough, it's fine for me.

With the version number.

> 7.91+dfsg1+really7.80+dfsg1-2
